crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -615,70 +616,113 @@ impl<F: Clone> PostUpdateTracker<F, PaymentData<F>, types::SdkPaymentsSessionUpd
         #[cfg(feature = "dynamic_routing")] _routable_connector: Vec<RoutableConnectorChoice>,
         #[cfg(feature = "dynamic_routing")] _business_profile: &domain::Profile,
     ) -> RouterResult<PaymentData<F>>
     where
         F: 'b + Send,
     {
-        // let session_update_details =
-        //     payment_data
-        //         .payment_intent
-        //         .tax_details
-        //         .clone()
-        //         .ok_or_else(|| {
-        //             report!(errors::ApiErrorResponse::InternalServerError)
-        //                 .attach_printable("missing tax_details in payment_intent")
-        //         })?;
-
-        // let pmt_amount = session_update_details
-        //     .pmt
-        //     .clone()
-        //     .map(|pmt| pmt.order_tax_amount)
-        //     .ok_or(errors::ApiErrorResponse::InternalServerError)
-        //     .attach_printable("Missing tax_details.order_tax_amount")?;
-
-        // let total_amount = MinorUnit::from(payment_data.amount) + pmt_amount;
-
-        // // if connector_ call successful -> payment_intent.amount update
-        // match router_data.response.clone() {
-        //     Err(_) => (None, None),
-        //     Ok(types::PaymentsResponseData::SessionUpdateResponse { status }) => {
-        //         if status == SessionUpdateStatus::Success {
-        //             (
-        //                 Some(
-        //                     storage::PaymentAttemptUpdate::IncrementalAuthorizationAmountUpdate {
-        //                         amount: total_amount,
-        //                         amount_capturable: total_amount,
-        //                     },
-        //                 ),
-        //                 Some(
-        //                     storage::PaymentIntentUpdate::IncrementalAuthorizationAmountUpdate {
-        //                         amount: pmt_amount,
-        //                     },
-        //                 ),
-        //             )
-        //         } else {
-        //             (None, None)
-        //         }
-        //     }
-        //     _ => Err(errors::ApiErrorResponse::InternalServerError)
-        //         .attach_printable("unexpected response in session_update flow")?,
-        // };
-
-        // let _shipping_address = payment_data.address.get_shipping();
-        // let _amount = payment_data.amount;
+        let connector = payment_data
+            .payment_attempt
+            .connector
+            .clone()
+            .ok_or(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("connector not found")?;
+
+        // For PayPal, if we call TaxJar for tax calculation, we need to call the connector again to update the order amount so that we can confirm the updated amount and order details. Therefore, we will store the required changes in the database during the post_update_tracker call.
+        if payment_data.should_update_in_post_update_tracker() {
```

Comment:
why is this defined on payment data. How can payment data know where to update? This information will be and should be known only by tax details struct. Is this also dependent on the connector?